lib/util/dump-collection.ts

```diff
@@ -0,0 +1,337 @@
+import ndjson from 'ndjson';
+import { JSONObject } from '../../index';
+import get from 'lodash/get';
+import isObject from 'lodash/isObject';
+import stream from 'stream';
+
+/**
+ * Flatten an object transform:
+ * {
+ *  title: "kuzzle",
+ *  info : {
+ *    tag: "news"
+ *  }
+ * }
+ *
+ * Into an object like:
+ * {
+ *  title: "kuzzle",
+ *  info.tag: news
+ * }
+ *
+ * @param {Object} target the object we have to flatten
+ * @returns {Object} the flattened object
+ */
+function flattenObject (target: JSONObject): JSONObject {
+  const output = {};
+
+  flattenStep(output, target);
+
+  return output;
+}
+
+function flattenStep (
+  output: JSONObject,
+  object: JSONObject,
+  prev: string | null = null): void {
+  const keys = Object.keys(object);
+
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    const value = object[key];
+    const newKey = prev ? prev + '.' + key : key;
+
+    if (Object.prototype.toString.call(value) === '[object Object]') {
+      output[newKey] = value;
+      flattenStep(output, value, newKey);
+    }
+
+    output[newKey] = value;
+  }
+}
+
+
+/**
+ * An iteration-order-safe version of lodash.values
+ *
+ * @param object The object containing the values
+ * @param fields The field names to pick in the right order
+ * @returns The values in the same order as the fields
+ * @see https://lodash.com/docs/4.17.15#values
+ */
+function pickValues (object: any, fields: string[]): any[] {
+  return fields.map(f => formatValueForCSV(get(object, f)));
+}
+
+/**
+ * Formats the value for correct CSV output, avoiding to return
+ * values that would badly serialize in CSV.
+ *
+ * @param value The value to format
+ * @returns The value or a string telling the value is not scalar
+ */
+function formatValueForCSV (value: any) {
+  if (isObject(value)) {
+    return '[NOT_SCALAR]';
+  }
+
+  return value;
+}
+
+abstract class AbstractDumper {
+  protected collectionDir: string;
+  protected options: {scroll: string, size: number};
+
+  protected abstract get fileExtension(): string;
+
+  constructor (
+    protected readonly index: string,
+    protected readonly collection: string,
+    protected readonly batchSize: number,
+    protected readonly query: any = {},
+    protected readonly writeStream: stream.Writable,
+  ) {
+    this.options = {
+      scroll: '2s',
```

Comment:
Will it only took 2sec to write the n documents from the batch in an http stream? This value was here because writing the filesystem is quite predictable but here I think we should have a bigger value, what do you think?